src/util-path.c

```diff
@@ -66,6 +66,35 @@ int PathIsRelative(const char *path)
     return PathIsAbsolute(path) ? 0 : 1;
 }
 
+/**
+ * Wrapper to join a directory and filename and resolve using realpath
+ *   _fullpath is used for WIN32
+ * @param out_buf output buffer.  Up to PATH_MAX will be written.
+ * @param buf_len length of output buffer
+ * @param dir the directory
+ * @param fname the filename
+ * @return TM_ECODE_OK on success.  On failure TM_ECODE_FAILED is returned.
+ */
+TmEcode PathJoin (char *out_buf, uint16_t buf_len, const char *const dir, const char *const fname)
+{
+    memset(out_buf, 0, buf_len);
+    uint16_t max_path_len = (buf_len > PATH_MAX) ? PATH_MAX : buf_len;
+    int bytes_written = snprintf(out_buf, max_path_len, "%s/%s", dir, fname);
```

Comment:
Can be done later, but I guess here we would use `\` for Windows?

Reply:
Good point, may as well do it now.